### ISSUE TITLE
Signing style in generated plist is automatic even though it was set to manual

### ISSUE THREAD

**Issue by m4rkk** [fastlane/fastlane#11059]
### New Issue Checklist

- [ x] Updated fastlane to the latest version
- [x ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x ] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [ x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Since upgrading to 2.67.0 manual signing is reverting to automatic, even though I set use_automatic_signing: false using the automatic_code_signing action.  Yesterday morning I did a build with no issues then later in the day there were code signing issues. No significant changes to my code so I am suspecting a recent fastlane change.

Here is the output I am seeing:

▸ -----------------------------------------
▸ {
▸   "provisioningProfiles": {
▸     "com.aliphcom.SurveyUS.internal.enterprise": "SurveyUS internal enterprise"
▸   },
▸   "method": "enterprise",
▸   "signingStyle": "automatic"
▸ }
▸ -----------------------------------------
This results in the following plist file:
▸ -----------------------------------------
▸ <?xml version="1.0" encoding="UTF-8"?>
▸ <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
▸ <plist version="1.0">
▸ <dict>
▸ 	<key>method</key>
▸ 	<string>enterprise</string>
▸ 	<key>provisioningProfiles</key>
▸ 	<dict>
▸ 		<key>com.aliphcom.SurveyUS.internal.enterprise</key>
▸ 		<string>SurveyUS internal enterprise</string>
▸ 	</dict>
▸ 	<key>signingStyle</key>
▸ 	<string>automatic</string>
▸ </dict>
▸ </plist>
▸ -----------------------------------------
[36m$ /usr/bin/xcrun /Users/cijenkins/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.67.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/mj/glp9xdk56m3_gwc1xf6t6vdmk5yfs7/T/gym_config20171128-4237-7qutb.plist' -archivePath /Users/cijenkins/Library/Developer/Xcode/Archives/2017-11-28/JHH\ Survey\ Dubai\ 2017-11-28\ 14.33.47.xcarchive -exportPath '/var/folders/mj/glp9xdk56m3_gwc1xf6t6vdmk5yfs7/T/gym_output20171128-4237-1w0ih3f' [0m
+ xcodebuild -exportArchive -exportOptionsPlist /var/folders/mj/glp9xdk56m3_gwc1xf6t6vdmk5yfs7/T/gym_config20171128-4237-7qutb.plist -archivePath '/Users/cijenkins/Library/Developer/Xcode/Archives/2017-11-28/JHH Survey Dubai 2017-11-28 14.33.47.xcarchive' -exportPath /var/folders/mj/glp9xdk56m3_gwc1xf6t6vdmk5yfs7/T/gym_output20171128-4237-1w0ih3f
2017-11-28 14:36:27.874 xcodebuild[6433:2676820] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/mj/glp9xdk56m3_gwc1xf6t6vdmk5yfs7/T/Survey Internal Dubai Enterprise Debug_2017-11-28_14-36-27.873.xcdistributionlogs'.
2017-11-28 14:36:29.313 xcodebuild[6433:2676820] [MT] IDEDistribution: Step failed: <IDEDistributionSigningAssetsStep: 0x7fa979db39a0>: Error Domain=IDEDistributionSigningAssetStepErrorDomain Code=0 "Locating signing assets failed." UserInfo={NSLocalizedDescription=Locating signing assets failed., IDEDistributionSigningAssetStepUnderlyingErrors=(
"Error Domain=IDEProfileLocatorErrorDomain Code=1 \"No profiles for 'com.aliphcom.SurveyDubai.internal.enterprise' were found\" UserInfo={NSLocalizedDescription=No profiles for 'com.aliphcom.SurveyDubai.internal.enterprise' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS In House provisioning profiles matching 'com.aliphcom.SurveyDubai.internal.enterprise'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}"
)}
error: exportArchive: No profiles for 'com.aliphcom.SurveyDubai.internal.enterprise' were found
Note that signingStyle is automatic even though I set it to manual.

### Captured Output

Command Used: `enterprise --env SurveyUS.internal.enterprise branch:release/1.1.0 build_number:999 is_release:false increment:false only_build:true allow_git_dirty:true`
<details><summary>Output/Log</summary>

```

Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
+------+------------------+-------------+

Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/mkieling/dev/ios-jhh/Health-Hub/fastlane/Appfile'
-------

Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `each' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60each'%20for%20nil+repo:fastlane/fastlane
➡️  Type Checking error: got 'Fixnum' for attribute 'fileEncoding' and undefined method `each' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/11016 [open] 1 💬
    a week ago

➡️  `fastlane deliver download_metadata` crash: NoMethodError: [!] undefined method `version' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/11046 [open] 6 💬
    9 hours ago

➡️  undefined method `[]' for nil:NilClass during Match
    https://github.com/fastlane/fastlane/issues/10960 [closed] 19 💬
    a week ago

and 111 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60each%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.


```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.1                                     |
| Ruby                        | 2.4.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.13.6 (Apple Git-96)           |
| Installation Source         | ~/.rvm/rubies/ruby-2.4.1/bin/fastlane       |
| Host                        | Mac OS X 10.13.1 (17B48)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.1/lib                |
| OpenSSL Version             | OpenSSL 1.1.0f  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# More documentation about how to customize your build
# can be found here:
# https://docs.fastlane.tools

#fastlane_version "2.60"
default_platform :ios

platform :ios do

	before_all do
		Dotenv.overload '.env.global'
		skip_docs
	end

	after_all do
	end

	private_lane :bump_increment do |options|
		puts "=== bump_increment"

		# Set build increment before building
		build_number = options[:build_number]
		current_build_number = get_build_number
		new_build_number = increment_build_number(build_number: build_number)

		puts "=== BUILD INCREMENT ==="
		puts "=== Current: #{current_build_number}"
		puts "=== Set To: #{new_build_number}"
	end

	private_lane :mangle_version do |options|
		puts "=== mangle_version"

		puts "=== Non release build so modify version to include the branch"
		current_version_number = get_version_number

		puts "=== VERSION NUMBER ==="
		puts "=== Current: #{current_version_number}"

		branch = options[:branch]

		if branch
			# Use some shell script to take out the slashes and dots then hand back the result
			branch = sh("temp=#{branch}; temp=${temp//\//-}; temp=${temp//./_}; printf $temp")
			new_version_number = "#{current_version_number}-#{branch}"
			new_version_number = increment_version_number(version_number: new_version_number)
			puts "=== Set To: #{new_version_number}"
		else
			puts "=== Expected a \"branch\" parameter"
		end
	end

	private_lane :build_app do |options|
		puts "=== build_app"

		automatic_code_signing(
 			path: "./" + ENV['XCODE_PROJECT'],
 			targets: ["#{ENV['XCODE_SCHEME']}"],
			use_automatic_signing: false
		)
		gym(
			scheme: ENV['XCODE_SCHEME'],
			clean: true,
			export_method: "enterprise",
			skip_profile_detection: true,
			export_options: {
				provisioningProfiles: {
					ENV['BUNDLE_ID'] => ENV['PROVISIONING_PROFILE_NAME']
				},
				method: "enterprise",
				signingStyle: "manual"
			}
		)
		automatic_code_signing(
 			path: "./" + ENV['XCODE_PROJECT'],
 			targets: ["#{ENV['XCODE_SCHEME']}"],
			use_automatic_signing: true
		)
	end

	private_lane :distribute_app do |options|
		puts "=== distribute_app"
		branch = options[:branch]

		# Add the sha1 hash to the notes so we know where the code lives in the repo
		commit = last_git_commit
		notes = options[:notes] + "\n*** Branch: " + branch + " ***"
		notes = notes + "\n*** Short commit hash: " + commit[:abbreviated_commit_hash] + " ***"
		notes = notes + "\n*** Bundle Identifier: " + ENV['BUNDLE_ID'] + " ***"
		notes = notes + "\n*** Provisioning Profile: " + ENV['PROVISIONING_PROFILE_NAME'] + " ***"

		# Distribution options
		emails = options[:emails]
		if emails == ""
			emails = nil
		end
		groups = options[:distro]
		if groups == "*none*" || groups == ""
			groups = nil
		end
		distribute = groups == nil && emails == nil ? false : true

		crashlytics(
			api_token: ENV['CRASHLYTICS_API_TOKEN'],
			build_secret: ENV['CRASHLYTICS_BUILD_SECRET'],
			crashlytics_path: ENV['CRASHLYTICS_PATH'],
			notes: notes,
			notifications: distribute,
			emails: emails,
			groups: groups
		)
	end

	private_lane :commit_increment_bump_to_git do |options|
		puts "=== commit_increment_bump_to_git"

		branch = options[:branch]

		if !branch
			puts "=== Expected a \"branch\" parameter"
		else
			puts "=== Cleaning up modified files except plists"
			clean_build_artifacts(
				exclude_pattern: '.*.Info\.plist'
			)
			puts "=== Committing version bumped plist files"
			new_build_number = get_build_number(xcodeproj: ENV['XCODE_PROJECT'])
			commit_version_bump(
				message: "Build Increment Bump: #{new_build_number}",
				xcodeproj: ENV['XCODE_PROJECT']
			)
			puts "=== Pushing version bump to remote"
			push_to_git_remote(
				local_branch: "HEAD",
				remote_branch: branch,
				tags: false
			)
		end
	end

	lane :enterprise do |options|

		if options[:allow_git_dirty] != true
			ensure_git_status_clean
		end
		clear_derived_data

		if options[:is_release] == true && options[:increment] == true
			bump_increment(options)
		end

		# For non release branch builds we will add the branch to the end of the version string making it easier to find in Fabric
		if options[:is_release] != true
			mangle_version(options)
		end		

		build_app(options)

		if options[:only_build] != true
			distribute_app(options)
			
			# Commit the version bump for release builds only
			if options[:is_release] == true && options[:increment] == true
				commit_increment_bump_to_git(options)
			end
		end

		if options[:allow_git_dirty] != true
			puts "=== Cleaning up git"
			reset_git_repo
		end
	end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# The Appfile can be used to specify information that's used across all fastlane
# tools, like your username or the app's bundle identifier.
#
# For more details, check out the documentation at:
# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

#for_lane :enterprise do
#  app_identifier "com.aliphcom.HealthHubEnterprise"
#  team_id ""
#end

#for_lane :testflight do
#	app_identifier "com.aliphcom.HealthHub"
#	team_id ""
#end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.67.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.2        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.15.3       |
| io-console                | 0.4.6        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.3.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.3        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.58.0       |
| faraday                   | 0.12.2       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
| json                      | 2.1.0        |
| openssl                   | 2.0.4        |
</details>

*generated on:* **2017-11-29**
</details>

**Comment by knellr** [comment]
We're also seeing this since 2.67.0. Reverting to 2.66.2 resolves the issue. In our case `signingStyle` is not set in Gym's generated plist:

```
        <key>signingStyle</key>
        <string></string>
```

**Comment by bobmccune** [comment]
Yes, I'm seeing this same behavior. The plist containing our export options has its signingStyle value set to 'manual', but fastlane is replacing this with an empty string.

This is a regression in 2.67.0 and the only valid workaround I've found is reverting to using 2.66.2.

**Comment by julestburt** [comment]
There has been an issue w/ manual code signing no longer getting set successfully on 9.x we found - there is still an open issue for this. So a workaround we managed was to set the project to manual, specifying the team and ensuring Dist is set for both release states.

#10637

**Comment by oronbz** [comment]
Yeah, same issue here. Worked perfectly before.

**Comment by oronbz** [comment]
Yo @ohayon, is this may be related: https://github.com/fastlane/fastlane/pull/11033?

**Comment by ohayon** [comment]
Hey @oronbz, yup. Definitely related. We are going to push out a fix today!

**Comment by kapfab** [comment]
FYI, this should be fixed soon thanks to PR #11217.

**Comment by kapfab** [comment]
I confirm this is now fixed in 2.70.0.

**Comment by KrauseFx** [comment]
That's awesome, thanks for confirming @kapfab 👍
